### PR TITLE
wait-for-job with backoff limit 0

### LINKED ISSUE #9285: --wait-for-jobs does not fails if job failed
Output of `helm version`: 
version.BuildInfo{Version:"v3.5.0", GitCommit:"32c22239423b3b4ba6706d450bd044baffdcf9e6", GitTreeState:"clean", GoVersion:"go1.15.6"}

Output of `kubectl version`:
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"archive", BuildDate:"2020-11-25T13:19:56Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.8", GitCommit:"9f2892aab98fe339f3bd70e3c470144299398ace", GitTreeState:"clean", BuildDate:"2020-08-13T16:04:18Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

I have this template:

```
apiVersion: batch/v1
kind: Job
metadata:
  name: backend-tests-job
  labels:
    app-name: tests
    app-service: backend-tests
spec:
  backoffLimit: 0
  template:
    metadata:
      name: backend-tests
    spec:
      restartPolicy: Never
      imagePullSecrets:
        - name: {{ .Values.imagePullSecrets }}

      containers:
      - name: tests
        image: {{.Values.image.repository}}/{{ .Values.appName }}:{{ .Values.appVersion }}
        imagePullPolicy: Always
        command: ["/usr/bin/make"]
        args: ["test", "PYTEST_ARGS=${ARGS}", "VERSION=${VERSION}"]
        env:
          - name: ARGS
            value: {{ .Values.pytestArgs | quote }}
          - name: GG_VERSION
            value: {{ .Values.version | quote }}
        resources:
            {{- toYaml .Values.resources | nindent 12 }}

```

I installs it with this command:
`helm install CHART --wait-for-jobs --wait --timeout 10m`

Sometimes job failed (because of tests, yeap :) ), does not restart (because of backoffLimit: 0) and then helm just... waits. 
I think that helm should not wait if job failed without any other restarts

### LINKED ISSUE #9312: --wait-for-jobs does not work with backOffLimit = 0
Using the new `--wait-for-jobs` flag, we encountered a bit of strange behavior when `backOffLimit` is set to zero. Setting the backoff to 0 as well as ` restartPolicy: Never` ensures that a job is only run once.

[The following check](https://github.com/helm/helm/blob/master/pkg/kube/wait.go#L193) validates failed runs are greater than or equal to 0. This condition will always return true when a job with `backOffLimit: 0` completes, because the `job.Status.Failed` will have a zero value (the field will not exist on job status).

### PR DESCRIPTION
**Pull request by tychedelia**
Because backOffLimit can be 0, a zero value for pod status failed will
always cause the condition to return true.

Signed-off-by: James McElwain <jmcelwain@gmail.com>

- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by yxxhero** [comment]
https://github.com/helm/helm/issues/9285 This PR will fix it.

**Comment by bacongobbler** [comment]
closing as a duplicate of #9285.

**Comment by jmcelwain** [comment]
@bacongobbler Is there another PR open for #9285? Happy if there's a more extensive fix out there, but this is the only one I see.

**Comment by bacongobbler** [comment]
Oh! My apologies. I thought @yxxhero was referencing another PR. I'll reopen this.

I need more coffee.

**Review by bacongobbler — APPROVED** [review]
LGTM. Thank you for writing the tests. Needs another maintainer's approval before we can merge.

**Comment by bacongobbler** [comment]
This PR also closes #9313.

**Comment by bacongobbler** [comment]
This PR also closes #9312.

**Review by jdolitsky — APPROVED** [review]

**Pull request merged**
